### PR TITLE
Feature/update dialog

### PR DESCRIPTION
**Pull request by two-one-five**
This adds an update dialog and renames the launcher in most places, if not all. But it needs further testing to confirm... some odd artifacts with the naming I saw but can't tell why.

This PR carries the commits from #12 as well, that one should be merged before this one.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -77,6 +77,7 @@
 
       <v-spacer></v-spacer>
 				<v-btn
+                    v-if="showDownloadButton"
```
Comment (ctrlaltdavid) [inline src/App.vue]: Indenting weird because of surrounding lines using tabs.

File: src/App.vue [hunk 2]
```diff
@@ -97,6 +98,17 @@
...
 				</v-btn>
+                
+                <v-btn
+                    v-if="showUpdateButton"
+                    v-on:click.native="checkForUpdates()"
+                    :right=true
+                    color="blue"
+                    :tile=true
+                >
+                    <span style="font-size: 12px;">Check for Updates</span>
```
Comment (ctrlaltdavid) [inline src/App.vue]: Should the styling be in the CSS file rather than inline?

File: src/App.vue [hunk 3]
```diff
@@ -428,22 +452,27 @@ export default {
 			shell.openExternal(url);
 		},
 		downloadInterface: function() {
+            // This function also auto-installs interface.
```
Comment (ctrlaltdavid) [inline src/App.vue]: Indenting / tabs.

File: src/App.vue [hunk 4]
```diff
@@ -458,7 +487,7 @@ export default {
         ipcRenderer.send('get-library-folder');
 	},
 	computed: {
-		interfaceSelected () {
+		interfaceSelected() {
```
Comment (ctrlaltdavid) [inline src/App.vue]: Tabs

File: src/background.js [hunk 5]
```diff
@@ -303,51 +303,52 @@ async function getLatestMetaJSON() {
         }
 	});
 	
-	var athenaMetaFile = storagePath.default + '/athenaMeta.json';
-	let rawdata = fs.readFileSync(athenaMetaFile);
-	let athenaMetaJSON = JSON.parse(rawdata);
+	var vircadiaMetaFile = storagePath.default + '/athenaMeta.json';
```
Comment (ctrlaltdavid) [inline src/background.js]: athenaMeta.json should be renamed to vircadiaMeta.json?

File: src/background.js [hunk 6]
```diff
@@ -303,51 +303,52 @@ async function getLatestMetaJSON() {
...
+	let rawdata = fs.readFileSync(vircadiaMetaFile);
+	let vircadiaMetaJSON = JSON.parse(rawdata);
 	
-	if (athenaMetaJSON) {
-		console.info("Athena Meta JSON:", athenaMetaJSON);
-		return athenaMetaJSON;
+	if (vircadiaMetaJSON) {
+		console.info("Athena Meta JSON:", vircadiaMetaJSON);
```
Comment (ctrlaltdavid) [inline src/background.js]: "Athena" occurrence.

File: src/background.js [hunk 7]
```diff
@@ -303,51 +303,52 @@ async function getLatestMetaJSON() {
...
+		return vircadiaMetaJSON;
 	} else {
         console.error("Failed to download Athena Meta JSON");
```
Comment (ctrlaltdavid) [inline src/background.js]: "Athena" occurrence.

File: src/background.js [hunk 8]
```diff
@@ -436,7 +437,7 @@ ipcMain.on('launch-interface', (event, arg) => {
 })
 
 ipcMain.on('get-athena-location', async (event, arg) => {
-	var athenaLocation = await getSetting('athena_interface.location', storagePath.interfaceSettings);
+	var athenaLocation = await getSetting('vircadia_interface.location', storagePath.interfaceSettings);
 	var athenaLocationExe = athenaLocation.toString();
 	console.info("AthenaLocationExe:",athenaLocationExe);
 	event.returnValue = athenaLocationExe;
```
Comment (ctrlaltdavid) [inline src/background.js]: "Athena" occurrences.

File: src/background.js [hunk 9]
```diff
@@ -637,13 +638,13 @@ async function silentInstall() {
...
 	var libraryPath;
-	var downloadURL = await shouldUpdate();
-    var installerName = "Athena_Setup_Latest.exe";
+	var downloadURL = await getDownloadURL();
+    var installerName = "Vircadia_Setup_Latest.exe";
     console.info("DLURL:", downloadURL);
 	if (downloadURL) {
-		getSetting('athena_interface.library', storagePath.default).then(function(results){
+		getSetting('vircadia_interface.library', storagePath.default).then(function(results){
 			if(results) {
```
Comment (ctrlaltdavid) [inline src/background.js]: Tabs

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by two-one-five** [comment]
Currently, this does not kick in unless there's an interface selected.

File: src/components/Dialogs/WantToClose.vue [hunk 10]
```diff
@@ -0,0 +1,71 @@
+<!--
+//  WantToClose.vue
+//
+//  Created by KasenVR on 19 Mar 2020.
+//  Copyright 2020 Project Athena and contributors.
```
Comment (ctrlaltdavid) [inline src/components/Dialogs/WantToClose.vue]:
```suggestion
//  Copyright 2020 Vircadia and contributors.
```

File: public/css/styles.css [hunk 11]
```diff
@@ -118,3 +132,11 @@ pre {
...
     background-color: #34455f !important;
 }
+
+#downloadButton span {
+    font-size: 12px;
+}
+
+#updateButton span {
+    font-size: 12px;
+}
```
Comment (ctrlaltdavid) [inline public/css/styles.css]: Add CR to end of file.

File: src/App.vue [hunk 12]
```diff
@@ -65,28 +75,43 @@
...
+			<v-btn
+                v-if="showDownloadButton"
+                v-on:click.native="downloadInterface()"
+                :right=true
+                color="blue"
+                :tile=true
+                :depressed="isDownloading"
+                :disabled="isSilentInstalling || disableDownloadButton"
+                id="downloadButton"
+			>
```
Comment (ctrlaltdavid) [inline src/App.vue]: Indentation.

File: src/App.vue [hunk 13]
```diff
@@ -411,44 +460,54 @@ export default {
 			shell.openExternal(url);
 		},
 		downloadInterface: function() {
-			if (!this.isDownloading) {
+            // This function also auto-installs interface.
```
Comment (ctrlaltdavid) [inline src/App.vue]:
```suggestion
            // This function also auto-installs Interface.
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by ctrlaltdavid** [comment]
Note: There are other files still containing "Athena", plus the execuables' icons need updating.

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by two-one-five** [comment]
I'm thinking we want to disable the launch button while an installation is running, I know the launcher is far more powerful (the ability to have multiple interfaces) but the sake of ease of use for the timebeing for such a trade off does not seem so bad.

**Pull request merged**